src/condor_procapi/procapi.cpp

```diff
@@ -1907,8 +1907,89 @@ ProcAPI::initProcInfoRaw(procInfoRaw& procRaw){
  */
 
 #if !defined(Darwin) && !defined(CONDOR_FREEBSD)
+
+#include <fstream>
+
 int
 build_pid_list( std::vector<pid_t> & newPidList ) {
+	//
+	// If you mount /proc with the option hidepid=2, then PID 1 isn't
+	// necessarily listed in /proc.
+	//
+	// See condor_utils/filesystem_remap.cpp for details.
+	//
+	static bool hidepid = true;
+	static bool checked_proc_mountinfo = false;
+	if(! checked_proc_mountinfo) {
+		std::string line;
+		std::ifstream file("/proc/self/mountinfo");
+		if( file.good() ) {
```

Comment:
Is `good()` sufficient to determine that the file exists and is readable?

Reply:
I believe so.  `ios::good()` is [defined](https://www.cplusplus.com/reference/ios/ios/good/) to mean that `eofbit`, `failbit`, and `badbit` are all not set.  The `ifstream()` [constructor](https://www.cplusplus.com/reference/fstream/ifstream/ifstream/) sets `badbit` if it couldn't open the file.

Reply:
Sounds good.